integration-tests/manifests/annotations/validate_annotations_namespace_test.go

```diff
@@ -186,22 +186,38 @@ func TestAutoAnnotationForManualAnnotationRemoval(t *testing.T) {
 	if err != nil {
 		t.Error("Error:", err)
 	}
-
-	startTime := time.Now()
+	startTime = time.Now()
 	updateTheOperator(t, clientSet, string(jsonStr))
 	if err != nil {
 		t.Errorf("Failed to get deployment app: %s", err.Error())
 	}
 
-	deployment, err := clientSet.AppsV1().Deployments(uniqueNamespace).Get(context.TODO(), deploymentName, metav1.GetOptions{})
-	if err != nil {
-		fmt.Printf("Error retrieving deployment: %v\n", err)
-		os.Exit(1)
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+
+	for {
+		deployment, err := clientSet.AppsV1().Deployments(uniqueNamespace).Get(ctx, deploymentName, metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				t.Fatalf("Deployment %s not found in namespace %s\n", deploymentName, uniqueNamespace)
+			}
+			t.Fatal("Error getting deployment")
+		}
+
+		if deployment.Status.AvailableReplicas == *deployment.Spec.Replicas && deployment.Status.UpdatedReplicas == *deployment.Spec.Replicas {
+			if deployment.Status.Replicas == deployment.Status.AvailableReplicas {
+				fmt.Println("All pods are fully ready and no pods are terminating.")
+				break
+			}
+		}
+
+		// Sleep for a short interval before checking again
+		time.Sleep(5 * time.Second)
```

Comment:
Should we be doing wait groups instead of sleep?

Reply:
Hmm I think wait groups might not be the best approach for this scenario because we're dealing with a loop that waits for a certain condition to be met, rather than coordinating multiple concurrent tasks.

Reply:
Wait groups might not be the most suitable approach for this scenario because we're dealing with a loop that waits for a certain condition to be met, rather than coordinating multiple concurrent tasks. It is a single goroutine that runs in a loop.